kroxylicious-integration-test-support/src/main/java/io/kroxylicious/test/tester/SimpleMetric.java

```diff
@@ -35,38 +34,39 @@ public record SimpleMetric(String name, Map<String, String> labels, double value
             .compile("^(?<metric>[a-zA-Z_:][a-zA-Z0-9_:]*)(\\{(?<labels>.*)})?[\\t ]*(?<value>[0-9E.]*)[\\t ]*(?<timestamp>\\d+)?$");
     private static final Pattern NAME_WITH_QUOTED_VALUE = Pattern.compile("^(?<name>[a-zA-Z_:][a-zA-Z0-9_:]*)=\"(?<value>.*)\"$");
 
+    private record LineMatcher(String line, Matcher matcher) {};
+
     static List<SimpleMetric> parse(String output) {
-        var all = new ArrayList<SimpleMetric>();
         try (var reader = new BufferedReader(new StringReader(output))) {
-            var line = reader.readLine();
-            while (line != null) {
-                if (!(line.startsWith("#") || line.isEmpty())) {
-                    var matched = PROM_TEXT_EXPOSITION_PATTERN.matcher(line);
-                    if (!matched.matches()) {
-                        throw new IllegalArgumentException("Failed to parse metric %s".formatted(line));
-                    }
-
-                    all.add(parseMetric(matched, line));
-                }
-                line = reader.readLine();
-            }
-            return all;
+            return reader.lines()
+                    .filter(line -> !(line.startsWith("#") || line.isEmpty()))
+                    .map(line -> new LineMatcher(line, PROM_TEXT_EXPOSITION_PATTERN.matcher(line)))
+                    .peek(SimpleMetric::verifyMatches)
+                    .map(SimpleMetric::parseMetric)
+                    .toList();
         }
         catch (IOException e) {
             throw new UncheckedIOException("Failed to parse metrics", e);
         }
     }
 
-    private static SimpleMetric parseMetric(Matcher matched, String line) {
+    private static void verifyMatches(LineMatcher lineMatcher) {
+        if (!lineMatcher.matcher.matches()) {
+            throw new IllegalArgumentException("Failed to parse metric %s".formatted(lineMatcher.line));
+        }
+    }
```

Comment:
If it was 

```suggestion
    private static boolean verifyMatches(LineMatcher lineMatcher) {
        return lineMatcher.matcher.matches()
    }
```

We would just be able to use it as a filter on the stream. Admittedly that makes it a silent failure but maybe thats ok.